controller/pkg/ipsetmanager/ipsets.go

```diff
@@ -0,0 +1,397 @@
+package ipsetmanager
+
+import (
+	"encoding/base64"
+	"io"
+	"net"
+	"sync"
+
+	ipsetpackage "github.com/aporeto-inc/go-ipset/ipset"
+	"github.com/spaolacci/murmur3"
+	provider "go.aporeto.io/trireme-lib/controller/pkg/aclprovider"
+
+	"go.aporeto.io/trireme-lib/policy"
+	"go.uber.org/zap"
+)
+
+const (
+	//IPv6DefaultIP is the default ip of v6
+	IPv6DefaultIP = "::/0"
+	//IPv4DefaultIP is the  default ip for v4
+	IPv4DefaultIP = "0.0.0.0/0"
+	//IPsetV4 version for ipv4
+	IPsetV4 = iota
+	//IPsetV6 version for ipv6
+	IPsetV6
+)
+
+//ACLManager interface is used by supervisor. This interface provides the supervisor to
+//create ipsets corresponding to service ID.
+type ACLManager interface {
+	// sets the ipset provider
+	SetIpsetProvider(ipset provider.IpsetProvider, ipsetVersion int)
+
+	AddToIPset(set provider.Ipset, data string) error
+	DelFromIPset(set provider.Ipset, data string) error
+
+	RegisterExternalNets(contextID string, extnets policy.IPRuleList) error
+	DestroyUnusedIPsets()
+	RemoveExternalNets(contextID string)
+	GetIPsets(extnets policy.IPRuleList, ipver int) []string
+}
+
+//IPsetUpdates is used by dns proxy currently to dynamically update the ips.
+type IPsetUpdates interface {
+	UpdateIPsets([]string, string)
+}
+
+type ipsetInfo struct {
+	contextIDs map[string]bool
+	name       string
+	addresses  map[string]bool
+}
+
+type handler struct {
+	serviceIDtoIPset      map[string]*ipsetInfo
+	contextIDtoServiceIDs map[string]map[string]bool
+	ipset                 provider.IpsetProvider
+	ipsetPrefix           string
+	ipFilter              func(net.IP) bool
+	ipsetParams           *ipsetpackage.Params
+	toDestroy             []string
+}
+
+type managerType struct {
+	ipv4Handler *handler
+	ipv6Handler *handler
+	sync.RWMutex
+}
+
+var manager managerType
+
+const (
+	ipv4String = "v4-"
+	ipv6String = "v6-"
+)
+
+//GetManager is called to get the object to manage acls.
+func GetManager() ACLManager {
+	return &manager
+}
+
```

Comment:
That's the part I don't want. We still have a package global variable. There should be an object created at the top that is passed to both enforcer and supervisor. We haven't avoided the complications of the package wide shared variables with this.